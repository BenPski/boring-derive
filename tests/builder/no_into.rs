use boring_derive::Builder;

#[derive(Debug, Default, Builder)]
struct Example {
    field1: f32,
    #[builder(no_into)]
    field2: String,
}

fn main() {
    let example = Example::default().field1(1.0).field2("something");
    println!("{:?}", example);
}
