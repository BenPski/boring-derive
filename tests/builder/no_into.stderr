error[E0308]: mismatched types
  --> tests/builder/no_into.rs:11:57
   |
11 |     let example = Example::default().field1(1.0).field2("something");
   |                                                  ------ ^^^^^^^^^^^- help: try using a conversion method: `.to_string()`
   |                                                  |      |
   |                                                  |      expected `String`, found `&str`
   |                                                  arguments to this method are incorrect
   |
note: method defined here
  --> tests/builder/no_into.rs:7:5
   |
3  | #[derive(Debug, Default, Builder)]
   |                          -------
...
7  |     field2: String,
   |     ^^^^^^
